keylime/registrar_client.py

```diff
@@ -131,7 +178,7 @@ def doRegisterAgent(
             logger.critical("Error: did not receive blob from Registrar Server: %s", response.status_code)
             return None
 
-        return response_body["results"]["blob"]
+        return cast(bytes, response_body["results"]["blob"])
```

Comment:
Uhm I am wondering why this cast. IIUC this comes from a `request.json()` that internally calls `loads`. If I read the code correctly it uses the Pyhon JSON module if `simplejson` is not installed. In my local installation, `simplejson` always return a `str` for `dumps`.

In any case I would return always `str`, as is more aligned with Python (for example, note that `dumps` returns a string, and `loads` cannot serialize `bytes`).

Reply:
Correct. This function has to return an `Optional[str]`. I now fixed the annotation of tpm_main's activate_identity() to also take a `str`.